dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java

```diff
@@ -490,23 +492,29 @@ public void setDescriptionAndDescriptionSource(String newDescription, Descriptio
         this.getVersionMetadata().descriptionSource = newDescriptionSource;
     }
 
-    // remove this method when author is removed from VersionMetadata
-    public void setAuthor(String newAuthor) {
-        this.getVersionMetadata().author = newAuthor;
-    }
-
-    // remove this method when author is removed from VersionMetadata
-    public void setEmail(String newEmail) {
-        this.getVersionMetadata().email = newEmail;
-    }
-
     public void addAuthor(final Author author) {
         boolean isNewAuthor = this.authors.stream().noneMatch(existingAuthor -> existingAuthor.getName().equals(author.getName()));
         if (isNewAuthor) {
             this.authors.add(author);
         }
     }
 
+    /**
+     * Sets the versions authors using authors from the descriptor.
+     * @param newAuthors
+     */
+    @JsonIgnore
+    public void setAuthorsFromDescriptor(Set<Author> newAuthors) {
```

Comment:
Why we need this function, rather than using `setAuthors`?

Reply:
It was an attempt to fix a bug where the previous Version .dockstore.yml authors weren't being cleared in the following scenario:
1. Workflow has descriptor authors and .dockstore.yml authors. The .dockstore.yml authors have higher precedence, so these are the Version's authors.
2. The .dockstore.yml authors are removed from the .dockstore.yml.
3. Expected behaviour is that the Version authors are the descriptor authors. Instead, they were a set of both descriptor and .dockstore.yml authors.

I found a better fix so I ended up removing this function